### PR TITLE
Separate DataTable from DataTableView

### PR DESCRIPTION
**Pull request by ajpal**
No visual changes here, just pulling the table itself into a separate React Component so that we can re-use it in the Preview Modal

Before:
image

After:
image

### CODE REVIEW

File: apps/src/storage/dataBrowser/DataTable.jsx [hunk 1]
```diff
@@ -374,9 +236,8 @@ class DataTable extends React.Component {
             label={msg.paginationLabel()}
           />
         </div>
-
-        <div style={styles.tableWrapper}>
-          <table style={tableDataStyle}>
+        <div>
```
Comment (jmkulwik) [inline apps/src/storage/dataBrowser/DataTable.jsx]: NIT: I think this div is no longer needed

File: apps/src/storage/dataBrowser/DataTableView.jsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+class DataTableView extends React.Component {
+  static propTypes = {
+    // from redux state
+    tableColumns: PropTypes.arrayOf(PropTypes.string).isRequired,
+    tableName: PropTypes.string.isRequired,
+    // "if all of the keys are integers, and more than half of the keys between 0 and
+    // the maximum key in the object have non-empty values, then Firebase will render
+    // it as an array."
+    // https://firebase.googleblog.com/2014/04/best-practices-arrays-in-firebase.html
```
Comment (jmkulwik) [inline apps/src/storage/dataBrowser/DataTableView.jsx:74]: NIT: Might be worth adding a bit of context here. It's not quite clear what the comment is referring to (I think it's referring to tableName).

Comment (ajpal) [reply apps/src/storage/dataBrowser/DataTableView.jsx:74]: `tableRecords` - it explains why `tableRecords` is either an object or an array. Same comment as currently at https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/storage/dataBrowser/DataTable.jsx#L81

Comment (jmkulwik) [reply apps/src/storage/dataBrowser/DataTableView.jsx:74]: Ahh makes sense.

**Review by jmkulwik — APPROVED** [review]

**Pull request merged**